src/core/syscalls/os_syscall_handler.rs

```diff
@@ -1,2 +1,944 @@
-#[allow(unused)] // TODO: remove after using.
-struct OsSyscallHandler {}
\ No newline at end of file
+use std::any::Any;
+use std::collections::{HashMap, VecDeque};
+
+use cairo_rs::types::relocatable::Relocatable;
+use cairo_rs::vm::vm_core::VirtualMachine;
+use cairo_rs::vm::vm_memory::memory_segments::MemorySegmentManager;
+
+use crate::core::errors::syscall_handler_errors::SyscallHandlerError;
+
+#[derive(Debug, Clone, PartialEq)]
+struct CallInfo {
+    caller_address: u32,
+    contract_address: u32,
+    internal_calls: Vec<CallInfo>,
+    entry_point_type: Option<EntryPointType>,
+    _storage_read_values: VecDeque<u32>, // U32
+    retadata: VecDeque<u32>,
+}
+
+impl Default for CallInfo {
+    fn default() -> Self {
+        Self {
+            caller_address: 0,
+            contract_address: 0,
+            internal_calls: Vec::new(),
+            entry_point_type: Some(EntryPointType::Constructor),
+            _storage_read_values: VecDeque::new(),
+            retadata: VecDeque::new(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+enum EntryPointType {
+    External,
+    L1Handler,
+    Constructor,
+}
+
+#[derive(Debug)]
+struct OsSingleStarknetStorage;
+
+#[derive(Debug, PartialEq)]
+pub struct TransactionExecutionInfo;
+
+impl OsSingleStarknetStorage {
+    // Writes the given value in the given key in ongoing_storage_changes and returns the
+    // previous value. This value is needed to create the DictAccess while executing the
+    // corresponding storage_write system call.
+    fn write(&self, _key: u32, _value: u32) -> u32 {
+        // TO BE IMPLEMENTED
+        todo!()
+    }
+}
+
+#[derive(Debug)]
+struct OsSyscallHandler {
+    tx_execution_info_iterator: VecDeque<TransactionExecutionInfo>,
+    call_iterator: VecDeque<CallInfo>,
+
```

Comment:
The struct should implement the SyscallHandler Trait, that would require this fields to be turn into RefCells because of the immutability.